app/src/main/java/com/hajduk/systems/prepareordermailing/woocommerce/WooCommerceClient.kt

```diff
@@ -0,0 +1,74 @@
+package com.hajduk.systems.prepareordermailing.woocommerce
+
+import com.github.kittinunf.fuel.core.Request
+import com.github.kittinunf.fuel.core.ResponseDeserializable
+import com.github.kittinunf.fuel.core.isSuccessful
+import com.github.kittinunf.fuel.httpGet
+import com.google.gson.FieldNamingPolicy
+import com.google.gson.GsonBuilder
+import com.hajduk.systems.prepareordermailing.woocommerce.exception.WooCommerceAuthorizationException
+import com.hajduk.systems.prepareordermailing.woocommerce.exception.WooCommerceResourceAccessException
+import com.hajduk.systems.prepareordermailing.woocommerce.model.Order
+
+class WooCommerceClient(
+    private val serverUrl: String,
+    private val clientKey: String,
+    private val clientSecret: String
+) {
+    companion object {
+        private const val O_AUTH_CONSUMER_KEY_PARAM_NAME = "oauth_consumer_key"
+        private const val O_AUTH_CONSUMER_TOKEN_PARAM_NAME = "oauth_consumer_token"
+        private const val O_AUTH_SIGNATURE_METHOD_PARAM_NAME = "oauth_signature_method"
+        private const val O_AUTH_SIGNATURE_METHOD = "HMAC-SHA1"
+        private const val O_AUTH_VERSION_PARAM_NAME = "oauth_version"
+        private const val O_AUTH_VERSION = "1.0"
+
+        private const val BASE_URL = "/wp-json/wc/v3"
+        private const val ORDERS_RESOURCE = "/orders"
+    }
+
+    fun getOrder(orderId: String): Order? {
+        return executeCall(buildUrl("$ORDERS_RESOURCE/$orderId").httpGet(), Order::class.java)
+    }
+
+    private fun buildUrl(resource: String): String {
+        return "$serverUrl$BASE_URL$resource" +
+                "?$O_AUTH_CONSUMER_KEY_PARAM_NAME=$clientKey" +
+                "&$O_AUTH_CONSUMER_TOKEN_PARAM_NAME=$clientSecret" +
+                "&$O_AUTH_SIGNATURE_METHOD_PARAM_NAME=$O_AUTH_SIGNATURE_METHOD" +
+                "&$O_AUTH_VERSION_PARAM_NAME=$O_AUTH_VERSION"
+    }
+
+    private fun <T : Any> executeCall(request: Request, responseClazz: Class<T>): T? {
+        val callResult = request.responseObject(GsonDeserializer(responseClazz))
+        when {
+            callResult.second.isSuccessful -> {
+                return callResult.third.component1()
+            }
+            callResult.second.statusCode == 404 -> {
+                return null
```

Comment:
poszukam jak obsługiwać wyjątki na froncie i myślę że tu też będzie trzeba palnąć wyjątek, co nam po tym nullu

Reply:
To owalimy statusami